violet-framework/src/test/java/com/horstmann/violet/framework/util/nodeusage/NodeUsagesFinderTest.java

```diff
@@ -0,0 +1,129 @@
+package com.horstmann.violet.framework.util.nodeusage;
+
+import com.horstmann.violet.product.diagram.abstracts.node.INode;
+import com.horstmann.violet.product.diagram.property.text.LineText;
+import com.horstmann.violet.product.diagram.property.text.SingleLineText;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.catchThrowable;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class NodeUsagesFinderTest
+{
+    private final NodeUsagesFinder nodeUsagesFinder = new NodeUsagesFinder();
+
+    private INode firstNode;
+    private INode secondNode;
+    private INode thirdNode;
+    private INode fourthNode;
+
+    private final List<INode> selectedNodes = new ArrayList<>();
+    private final List<INode> allNodes = new ArrayList<>();
+
+    @Before
+    public void Set_Up()
+    {
+        firstNode = mock(INode.class);
+        when(firstNode.getName()).thenReturn(asLineText("First"));
+        when(firstNode.getAttributes()).thenReturn(asLineText(""));
+        when(firstNode.getMethods()).thenReturn(asLineText(""));
+
+        secondNode = mock(INode.class);
+        when(secondNode.getName()).thenReturn(asLineText("Second"));
+        when(secondNode.getAttributes()).thenReturn(asLineText("first : First \n third : Third"));
+        when(secondNode.getMethods()).thenReturn(asLineText(""));
+
+        thirdNode = mock(INode.class);
+        when(thirdNode.getName()).thenReturn(asLineText("Third"));
+        when(thirdNode.getAttributes()).thenReturn(asLineText(""));
+        when(thirdNode.getMethods()).thenReturn(asLineText("doSomething(first : First) : First"));
+
+        fourthNode = mock(INode.class);
+        when(fourthNode.getName()).thenReturn(asLineText(null));
+        when(fourthNode.getAttributes()).thenReturn(asLineText(null));
+        when(fourthNode.getMethods()).thenReturn(asLineText(null));
+
+    }
+
+    @Test
+    public void Should_Throw_Exception_When_Parameters_Are_Null()
```

Comment:
I think it's ok - if the name is long, we can use '_' instead of camelCase. 🐫 It improves readability.
DoIHaveRightYea?TheLongNamesArentAlwaysGoodAndTheBest :P